php_phongo.c

```diff
@@ -1878,17 +1264,98 @@ static bool php_phongo_apply_wc_options_to_uri(mongoc_uri_t *uri, bson_t *option
 	return true;
 } /* }}} */
 
+static bool php_phongo_apply_ssl_opts(mongoc_client_t *client, zval *zdriverOptions TSRMLS_DC)
+{
+	mongoc_ssl_opt_t ssl_opt = {0};
+	zval *zoptions;
+	zend_bool free_pem_file = 0, free_pem_pwd = 0, free_ca_file = 0, free_ca_dir = 0;
+	int plen;
+
+	if (!zdriverOptions) {
+		return false;
+	}
+
+	zoptions = zdriverOptions;
+
+	if (php_array_existsc(zdriverOptions, "context")) {
+		php_stream_context *context;
+		zval *zcontext;
+
+		zcontext = php_array_fetchc(zdriverOptions, "context");
+
+		/* TODO: Do we want to check global stream options, too? */
+		context = php_stream_context_from_zval(zcontext, 1);
```

Comment:
I've opted to ignore the global stream context. Leaving this code as-is, since the second argument means we return `NULL` instead of defaulting to the global context.